### PR TITLE
[S] Illegal Paper-Tech Fix (GIBS... IN MY UPLINK?!!)

### LINKED ISSUE #39159: Paper is highly illegal.
[Round ID]: #91066

[Testmerges]:
#39132: 'Stops various machines from processing needlessly' by vuonojenmustaturska at commit f1be8a8c4a
#39139: 'Makes damage precision explicit' by AnturK at commit faca150c53
#39135: 'Revert "You can no longer SDQL select certain datums"' by MrStonedOne at commit e289efa87b
#39111: 'Custom names refactor' by AnturK at commit c029b33f05

Round ID: 91066

[Reproduction]: # 
Grab yourself some ordinary paper out the paper bin.
Shove it in the DA.
UNLOCK ILLEGAL TECHNOLOGY BECAUSE PAPER IS THE PRODUCT OF THE SYNDICATE.

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl: Cobby
fix: Paper doesn't give illegal tech anymore
/:cl:

fixes #39159

Trying to find something trash that also wasn't a common item is pretty difficult, you should try it sometime when you're bored.

### CODE REVIEW

**Comment by XDTM** [comment]
sords can be found in lavaland

**Comment by XDTM** [comment]
Some cosmetic traitor or nuke op base item might do the trick, maybe a traitor headset

**Comment by ExcessiveUseOfCobblestone** [comment]
https://github.com/tgstation/tgstation/blob/4fe456d744010bc3dae4af6c4b99ecb2b6919326/code/modules/mining/lavaland/necropolis_chests.dm#L13-L79

can you show me where? I know it used to have a sord but can't seem to find it anymore.

**Comment by ExcessiveUseOfCobblestone** [comment]
Thanks @XDTM

**Comment by Nabski89** [comment]
Can't you find syndicate balloons in delta maint?

**Comment by ExcessiveUseOfCobblestone** [comment]
dunno i don't play delta

**Comment by nicbn** [comment]
Create a new item for it like `syndicatebaloon/random`

**Comment by ExcessiveUseOfCobblestone** [comment]
creating a subtype for the explicit purpose of cockblocking techwebs seems very silly, I just used something that isn't tangible to trip the system.

Doubles as an obvious "HEY SOMETHING WENT WRONG"

**Comment by Jared-Fogle** [comment]
@Nabski89 You can buy syndicate balloons from the uplink, so that's not something you can fix unless you remove those from maint.

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -1553,7 +1553,7 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 /datum/uplink_item/badass/random
 	name = "Random Item"
 	desc = "Picking this will purchase a random item. Useful if you have some TC to spare or if you haven't decided on a strategy yet."
-	item = /obj/item/paper
+	item = /obj/effect/gibspawner/generic // non-tangible item because techwebs use this path to determine illegal tech
```
Comment (Jalleo) [inline code/modules/uplink/uplink_items.dm]: someone listened to me wew

**Comment by Supermichael777** [comment]
Sords are so dumb their illegal, no issues

File: code/modules/uplink/uplink_items.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/uplink/uplink_items.dm]: oh lol

**Comment by AnturK** [comment]
Why do these even have a path in there or alternatively why it's not handled by R&D code ?

**Comment by ExcessiveUseOfCobblestone** [comment]
@AnturK when establishing the list of uplink items it checks for `ispath(item)` or something similar.

I could also remove the check (the only time this would occur would be in a PR diff) and remove the item path, but with this being such an indirect exploit I wanted to make sure I didn't cause even more indirect problems than currently.

As for RND, I dunno. Not sure why it's based on the uplink and not on the object itself (time for a research bonus component???).

**Comment by AnturK** [comment]
I'd prefer if these didn't have any absolutely irrelevant data attached, be it through additional variable for uplink items that override item creation or some change to RND code.

**Comment by ExcessiveUseOfCobblestone** [comment]
I can look into it for a more proper solution but for the purpose of people not using paper for illegal tech this might be good to testmerge until then.

**Comment by ExcessiveUseOfCobblestone** [comment]
i'm going to have to bulldoze uplink_items.dm ARE YOU HAPPY

**Review by Cyberboss — APPROVED** [review]

**Comment by ShizCalev** [comment]
Wait, shit I just though of something. What does this show up as in the round end report when it's purchased?

**Comment by ExcessiveUseOfCobblestone** [comment]
please tell me it's gibs

**Pull request merged**